### ISSUE TITLE
Add support for `apollo`

### ISSUE THREAD

**Issue by BKLronin** [openandroidinstaller-dev/openandroidinstaller#490]
### Is your device already requested?

- [X] I have searched the existing issues

### Brand

Xiaomi 

### Name

Mi10 T PRO

### Device code

apollo

### Device images

```Markdown
https://download.lineageos.org/devices/apollon/builds
```

### Do you own the device and would be willing to test the installer?

- [X] I own the device and am willing to test the installer on it.

### Additional context

```Markdown
Needs to be unlocked by the manufacturer first but after that it would be nice to update with androidinstaller.
```

**Comment by anon1892** [comment]
Hello, thanks for your interest in this project!
I've just looked at LineageOS installation process and created a configuration file for this device. #493 
Since it's not tested yet, could you help us by testing it please? We can help you if needed.

**Comment by BKLronin** [comment]
Thank you for providing this. :)

I could try, Its my main phone so I need to be a bit careful. Iḿ already on Lineage 19.1 so things might work a little different.  

Instructions would be good. Iḿ on manjaro or arm64 Sonoma mac.

**Comment by tsterbak** [comment]
Hey, the installer will do a clean install with a factory reset, so you should backup everything you want to keep.
Besides from that, it will use TWRP to sideload the ROM, so usually it is basically impossible to brick the device if you know how to flash a recovery by cli if necessary.

The easiest way to test it is to use the linux executable and put the config file from the PR to the directory the executable is running from. The installer will pick this up and support your device. Alternatively, you could do the python setup with poetry and run the installer from there.

Let me know if you need any help :)

**Comment by BKLronin** [comment]
I flashed manually yesterday as I dont want a factory reset.
If the internal updater cant patch to the next major version, would it make sense to add a second script that patches phones with already existing lineage oses installed?

**Comment by anon1892** [comment]
I don't think so: if LOS team didn't allow Android versions changes directly from their updater, it's because it isn't without risk. The success of dirty flashing highly depends on the device and android version, and there is no data for this.

**Comment by BKLronin** [comment]
It is the official strategy for an upgrade from say 19. to 20. https://wiki.lineageos.org/devices/apollon/upgrade/variant1/

Maybe a different technical reason why its not done from the internal updater.

If theres no checkbox for "already on lineage" the average user, this installer is aimed at, would be stuck.

**Comment by anon1892** [comment]
This guide is supposed to work with LOS recovery, which is not currently supported by OpenAndroidInstaller (and probably not for a long time, since it does not support all `adb shell twrp ...` useful commands for scripting...
And I've heard many stories of dirty flash procedures that didn't work, I think it's risky to add it into OAI as a second script. But maybe we can add an option in the future "Advanced Mode" to not format `data` partition... What do you think @tsterbak ?

**Comment by BKLronin** [comment]
Ah I see, didn't noticed the different recovery. 

At least for this specific case I can report that flashing went successful without any losses.

**Comment by tsterbak** [comment]
> And I've heard many stories of dirty flash procedures that didn't work, I think it's risky to add it into OAI as a second script. But maybe we can add an option in the future "Advanced Mode" to not format data partition... What do you think @tsterbak ?

@anon1892 I never do dirty flashes, since I had bad experiences with it and I don't think it is encouraged at all. But once we scope the "Advanced Mode", we could add an option to skip factory resets and partition wipes to enable dirty flashes.

> At least for this specific case I can report that flashing went successful without any losses.

@BKLronin You mean dirty flashing, right?

Also, do you still see an option to test the config with your device? :)

**Comment by tsterbak** [comment]
You could open an Issue to request options to upgrade Android versions and/or dirty flash :) this way it is not lost in this device specific issue and I can add it to the road map.

**Comment by BKLronin** [comment]
> Also, do you still see an option to test the config with your device? :)

I´m afraid not at the moment but I can offer to add devices in the future If I get a new one or lost device etc. 
I´didn´t have in mind that it requires a wipe when adding the device on the wishlist.

I have a BV6000 that is free for testing but its not very popular or widespread.
